src/main/scala/viper/server/vsi/VerificationServerInterface.scala

```diff
@@ -0,0 +1,310 @@
+        task.setQueueActor(message_actor)
+        val task_with_actor = new Thread(task)
+        val answer = job_actor ? VerificationProtocol.Verify(task_with_actor, queue, publisher)
+        val new_job_handle: Future[JobHandle] = answer.mapTo[JobHandle]
+        new_job_handle
+      }
+      val (id, _) = jobs.bookNewJob(createJob)
+      VerificationJobHandler(id)
+    } else {
+      println(s"the maximum number of active verification jobs are currently running ($jobs.MAX_ACTIVE_JOBS).")
+      VerificationJobHandler(-1) // Not able to create a new JobHandle
+    }
+  }
+
+  protected def stop(): Unit = {
+    getInterruptFutureList() onComplete {
+      case Success(_) =>
+        _termActor ! Terminator.Exit
+        println(s"shutting down...")
+      case Failure(err_msg) =>
+        println(s"Interrupting one of the verification threads timed out: $err_msg")
+        _termActor ! Terminator.Exit
+        println(s"forcibly shutting down...")
+    }
+  }
+
+  /** This method interrupts active jobs upon termination of the server.
+    */
+  protected def getInterruptFutureList(): Future[List[String]] = {
+    val interrupt_future_list: List[Future[String]] = jobs.jobHandles map { case (jid, handle_future) =>
+      handle_future.flatMap {
+        case JobHandle(actor, _, _) =>
+          implicit val askTimeout: Timeout = Timeout(1000 milliseconds)
+          (actor ? VerificationProtocol.Stop(true)).mapTo[String]
+      }
+    } toList
+    val overall_interrupt_future: Future[List[String]] = Future.sequence(interrupt_future_list)
+    overall_interrupt_future
+  }
+
+  protected def successHandleCallback(handle: JobHandle, clientActor: ActorRef)
+
+  /** A verification process ends after the results are retrieved.
+    *
+    * This should be done providing an actor that can receive the envelopes stored in the Queue actor's source queue
+    */
+  protected def terminateVerificationProcess(jid: Int, clientActor: ActorRef): Unit ={
+    jobs.lookupJob(jid) match {
+      case Some(handle_future) =>
+        handle_future.onComplete({
+          case Success(handle) =>
+            successHandleCallback(handle, clientActor)
+            _termActor ! Terminator.WatchJobQueue(jid, handle)
+          case Failure(e) =>  clientActor ! e
+        })
+      case None => clientActor ! JobNotFoundException()
+    }
+  }
+}
+
+
+/** This trait is a generic wrapper for a any sort of verification a VerificationServer might
+  * work on.
+  *
+  * It has the following properties:
+  *  - implements runnable
+  *  - provides a reference to a queue actor.
+  *
+  *  The first serves the purpose of running the process concurrently. The second allows to
+  *  communicate from the verification process to the server.
+  * */
+abstract class VerificationTask()(implicit val executionContext: ExecutionContext) extends Runnable {
+
+  protected var q_actor: ActorRef = _
+
+  def setQueueActor(actor: ActorRef): Unit = q_actor = actor
```

Comment:
For methods that have side effects, please use curly brackets to define their body. (I think that's the common Scala convention). Here I would write: 
```Scala 
def setQueueActor(actor: ActorRef): Unit = {
    q_actor = actor
}
```